### ISSUE TITLE
connectivity.cloudflareclient.com IP Addresses

### ISSUE THREAD

**Issue by jamie-sandbox** [cloudflare/cloudflare-docs#16062]
### Existing documentation URL(s)

https://developers.cloudflare.com/cloudflare-one/connections/connect-devices/warp/deployment/firewall/#connectivity-check

### What changes are you suggesting?

The documentation states the following:

> “Because this check happens inside of the tunnel, you do not need to add `connectivity.cloudflareclient.com` to your firewall allowlist.”

This is not necessarily correct, since on a Windows system with a firewall policy where outbound traffic is blocked by default, a rule must be added to allow `warp-svc.exe` to generate outbound network traffic to `connectivity.cloudflareclient.com`. 

The hostname `connectivity.cloudflareclient.com` currently resolves to `162.159.138.65` and `162.159.137.65`. However, these IPs are not referenced or contained elsewhere within the documentation.

Please can clarification be provided? Are `162.159.138.65` and `162.159.137.65` static addresses which we can create a firewall rule for? Or are they part of a range which we need to include the entirety of? If so, what is the range?

**Comment by jamie-sandbox** [comment]
Resolves to `162.159.138.65` and `162.159.137.65` when queried both inside and outside of the tunnel.

Hopefully these are static. :pray:

**Comment by deansundquist** [comment]
The client doesn't use DNS resolution for its inside tunnel connectivity checks, and it doesn't use `162.159.138.65` nor `162.159.137.65`.

**Comment by deansundquist** [comment]
@jamie-sandbox and @deadlypants1973 I apologize but my last comment was incorrect.  We do utilize DNS resolution for this, and we will attempt to connect to that IP address.  (I was thinking about the outside tunnel checks, which are handled differently).  

This traffic to this endpoint though should always be inside of the tunnel, when we connect to it.  If it is not, the client will not be able to connect (unless you disable connectivity checks).  

The reason we do not list this on our WARP with Firewall page is because it should be in the WARP tunnel, and not visible to any firewall that might sit between the Client and the Internet.  

If you see connections to connectivity.cloudflareclient.com outside of the tunnel, we'd suggest opening a support case, as this is unexpected.

**Comment by jamie-sandbox** [comment]
@deansundquist Thanks for your response.

From my original description:

> on a Windows system with a firewall policy where outbound traffic is blocked by default, a rule must be added to allow `warp-svc.exe` to generate outbound network traffic to `connectivity.cloudflareclient.com`.

Unfortunately the question still is not fully answered. Even when the traffic is within the tunnel, we still need to configure the local Windows Firewall to allow `warp-svc.exe` to send traffic to the relevant addresses.

The documentation should state what these addresses are.

**Comment by deadlypants1973** [comment]
https://github.com/cloudflare/cloudflare-docs/issues/16241#issuecomment-2670025678

**Comment by deadlypants1973** [comment]
@jamie-sandbox thank you for your reply. I have alerted the engineering team about this and they are looking into it. Will update!

**Comment by deadlypants1973** [comment]
Hey @jamie-sandbox , I responded here: https://github.com/cloudflare/cloudflare-docs/issues/16241#issuecomment-2787069354 Let me know if that answers your questions and I can close this out. Thank you!

**Comment by jamie-sandbox** [comment]
> Hey [@jamie-sandbox](https://github.com/jamie-sandbox) , I responded here: [#16241 (comment)](https://github.com/cloudflare/cloudflare-docs/issues/16241#issuecomment-2787069354) Let me know if that answers your questions and I can close this out. Thank you!

**Unfortunately this still isn't answered.** Traffic that goes through the tunnel is still passed through Windows Firewall. When outbound traffic is blocked by default in Windows Firewall, you need to permit `warp-svc.exe` to send traffic to the relevant addresses for the connectivity check.

Currently we have whitelisted `162.159.138.65` and `162.159.137.65` however it is not clear if these are truly static or not.

I've noticed that the documentation has been updated recently to add some new IPs to the connectivity requirements. These IPs seem to have been taken from the MASQUE range. I'm not sure of the context behind this and how/if it relates to this issue?

@ranbel @deadlypants1973  Please re-open.

**Comment by deadlypants1973** [comment]
Hi @jamie-sandbox , I think I understand where the issue is. I've included a docs update (https://github.com/cloudflare/cloudflare-docs/pull/23440) that makes clear the purpose of `connectivity.cloudflareclient.com`.  The 2 IPs that are required for connection to WARP are the ones listed in the [WARP with Firewall doc](https://developers.cloudflare.com/cloudflare-one/connections/connect-devices/warp/deployment/firewall/#inside-tunnel), not `connectivity.cloudflareclient.com`.

Additionally, [required scopes](https://developers.cloudflare.com/cloudflare-one/connections/connect-devices/warp/deployment/firewall/#inside-tunnel) covers that C:\Program Files\Cloudflare\Cloudflare WARP\warp-svc.exe needs to be allowed for Windows. Following these principles, everything should work as intended and you do not need to (and ideally should not) note the IPs of `connectivity.cloudflareclient.com`.

**Comment by jamie-sandbox** [comment]
@deadlypants1973 Thanks for the update and PR (#23440), but unfortunately I still don't think the question is answered.

The local Windows Firewall **does** need to permit the `443/tcp` connections to `connectivity.cloudflareclient.com`. Suggesting that this FQDN shouldn't be used in any firewall policy is incorrect.

Even though the connections go through the already-established tunnel, the local Windows Firewall still applies outbound filtering. I can absolutely confirm that not having these IPs allowed by outbound policy causes WARP to break. We have tested this. This is what caused me to log this issue in the first place.

The lack of clarity and confirmation is a constant worry that WARP could break at any minute because the IPs have changed from `162.159.137.65` and `162.159.138.65` (which are what `connectivity.cloudflareclient.com` currently resolve to).

I imagine that in most cases, this is not noticed as there would be a general rule to permit outbound `443/tcp` traffic from the endpoint, but in environments where this is restricted by default, we need to know the exact IPs to permit.

Please can this be further reviewed. I am happy to take a call with the engineering team to discuss directly if that'd be easier.

Thanks for your help.

**Comment by deadlypants1973** [comment]
Hi @jamie-sandbox , talked with eng and we see that in your setup you block outbound traffic by default. Our docs are written assuming customers allow outbound traffic by default and this may be where the issue is and would need docs clarification.

Could you help us confirm something?

If you only allow outbound traffic to the two IPs listed in our [documentation](https://developers.cloudflare.com/cloudflare-one/connections/connect-devices/warp/deployment/firewall/#inside-tunnel) (162.159.197.4 and 2606:4700:102::4) and remove any firewall rules involving connectivity.cloudflareclient.com's 162.159.137.65 and 162.159.138.65, does WARP stop functioning in your environment?

Thanks so much for your help and patience.

**Comment by jamie-sandbox** [comment]
@deadlypants1973 Thanks, let me test this for you, will reply back in a couple of days! 😄

**Comment by jamie-sandbox** [comment]
@deadlypants1973 Ok, sorry for the delay, got some results! 🥁 

On versions of WARP for Windows prior to `2025.4.929.0` (before 14th May 2025), removing `162.159.137.65` and `162.159.138.65` (`connectivity.cloudflareclient.com`) from the outbound Windows Firewall causes WARP to break (`CF_HOST_UNREACHABLE_CHECK`).

However, on version `2025.4.929.0` (14th May 2025) and higher, WARP is able to connect successfully. 🎉 

I notice that the changelog for `2025.4.929.0` says the following:

> _Added new IP addresses for tunnel connectivity checks. If your organization uses a firewall or other policies you will need to exempt these IPs._

Both of these tests were with the new connectivity check IPs (`162.159.197.3` and `162.159.197.4`) present in the outbound Windows Firewall.

So, I am glad this is working, but there really are still questions about what `connectivity.cloudflareclient.com` is used for, and what the new connectivity check IPs are.

a) The new connectivity check IPs are taken from the MASQUE range. I assume this is intentional?

b) Are these new IPs additional to the existing ones, or replacing the existing ones?

c) If `connectivity.cloudflareclient.com` is still used, the point still stands that we would need to know the IPs (and that they're static) in order to whitelist them on the outbound Windows Firewall (despite the fact they go through the tunnel).

I really appreciate everyone's help with this! 👍

**Comment by deadlypants1973** [comment]
Hi @jamie-sandbox , thank you for testing and sharing your results!

To answer your questions:

a) The new connectivity check IPs are taken from the MASQUE range. I assume this is intentional?
-> Yes, this is intentional. The connectivity check is protocol agnostic, so it works with MASQUE and WireGuard. It just so happens the IPs are from the MASQUE range.

b) Are these new IPs additional to the existing ones, or replacing the existing ones?
-> Replacing.

c) If connectivity.cloudflareclient.com is still used, the point still stands that we would need to know the IPs (and that they're static) in order to whitelist them on the outbound Windows Firewall (despite the fact they go through the tunnel).
-> DNS resolution is not used for `connectivity.cloudflareclient.com` in this context. Instead, WARP always overrides it with the specific connectivity IPs listed [in our documentation](https://developers.cloudflare.com/cloudflare-one/connections/connect-devices/warp/deployment/firewall/#connectivity-checks).

It’s great to hear that allowlisting those documented IPs works in your setup, that’s exactly the outcome we aimed for with the WARP release in April. Going forward, you only need to allow the IPs provided in the docs, as you’ve already done.

The connectivity IPs are static and are not expected to change.

**Comment by jamie-sandbox** [comment]
Thanks @deadlypants1973, makes sense, I'm glad we have gotten to the bottom of this. 😄

The key bit from your pull request (#23643) is:

> WARP overrides the resolved IPs with the IPs listed above.

Thanks again all for your help!

**Comment by kodster28** [comment]
Closed as the underlying PR is also closed
